### DISCUSSION TITLE
How to update parent span name with OTel java agent

### DISCUSSION THREAD

**Discussion by Raushan-Sapiens** [open-telemetry/opentelemetry-java-instrumentation#12764] [Q&A]
Hi,

We are writing [extensions ](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/4d6b1cdf3831f16e01e4b4082803e56c2bb11033/examples/extension)for SpanProcessor, and in the onStart [method](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4d6b1cdf3831f16e01e4b4082803e56c2bb11033/examples/extension/src/main/java/com/example/javaagent/DemoSpanProcessor.java#L30), we are trying to override the span name. But it looks like it only happens for child spans.

 Is there any way to update parent span name

Thanks

**Comment by trask** [comment]
hi @Raushan-Sapiens, I'm not quite certain what you're asking, but maybe this?

`Span.fromContext(parentContext).updateName("...")`
